client/controller/index.web.js

```diff
@@ -4,48 +4,52 @@
 import React from 'react';
 import ReactDom from 'react-dom';
 import { Provider as ReduxProvider } from 'react-redux';
-import { setSection as setSectionAction } from 'state/ui/actions';
-import noop from 'lodash/noop';
+import page from 'page';
 
 /**
- * Internal dependencies
+ * Internal Dependencies
  */
-import page from 'page';
+import Layout from 'layout';
 import LayoutLoggedOut from 'layout/logged-out';
+import layoutFocus from 'lib/layout-focus';
+import nuxWelcome from 'layout/nux-welcome';
+import translatorInvitation from 'layout/community-translator/invitation-utils';
+import { makeLayoutMiddleware } from './index.node.js';
 import { getCurrentUser } from 'state/current-user/selectors';
+import userFactory from 'lib/user';
+import sitesFactory from 'lib/sites-list';
 import debugFactory from 'debug';
 
-const debug = debugFactory( 'calypso:controller' );
-
 /**
- * @param { object } context -- Middleware context
- * @param { function } next -- Call next middleware in chain
-*/
-export function makeLayout( context, next ) {
-	const isLoggedIn = !! getCurrentUser( context.store.getState() );
-	if ( ! isLoggedIn ) {
-		context.layout = makeLoggedOutLayout( context );
-	} // TODO: else { makeLoggedInLayout( context ); }
-	next();
-}
+ * Re-export
+ */
+export { setSection } from './index.node.js';
```

Comment:
I find this a bit awkward, but I guess I could be confused in thinking of `.node.js` and `.web.js` as contextually exclusive, whereas this seems to treat the web variant as a superset of the "base" Node version. The downside I see in this is that all of the Node code will be pulled into the web bundle. Should functions like these have their placement reflect the fact that they're shared (i.e. separate file)?